macos/library/FLEViewController.m

```diff
@@ -265,29 +285,43 @@ + (FlutterRendererConfig)createRenderConfigHeadless:(BOOL)headless {
 }
 
 - (void)handlePlatformMessage:(const FlutterPlatformMessage *)message {
-  // This method must not return early as we still need to submit the
-  // response callback in case of error.
-
   NSData *data = [NSData dataWithBytesNoCopy:(void *)message->message
                                       length:message->message_size
                                 freeWhenDone:NO];
-
-  NSError *error = nil;
-  NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
-
   NSString *channel = @(message->channel);
-  id<FLEPlugin> plugin = _plugins[channel];
-  if (plugin != nil && dictionary != nil) {
-    id response = [plugin handlePlatformMessage:dictionary];
 
-    NSData *responseData = nil;
-    if (response != nil) {
-      responseData = [NSJSONSerialization dataWithJSONObject:response options:0 error:&error];
+  id messageObject = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+  FLEMethodCall *methodCall =
+      messageObject ? [FLEMethodCall methodCallFromMessage:messageObject] : nil;
+
+  __block const FlutterPlatformMessageResponseHandle *responseHandle = message->response_handle;
```

Comment:
It might be worth a comment explaining that responseHandle isn't just a convenient intermediate variable but is captured in the block below. (Maybe this is obvious to people who write Objective-C more regularly though.)